### PR TITLE
OHM-252 Metrics refactor

### PR DESCRIPTION
**Pull request by rcrichton**
@hnnesv could you please review this for me. This fixes the reporting issue and re implements the metrics API. It's a breaking change but I think this is for the better.

### CODE REVIEW

File: config/config.md [hunk 1]
```diff
@@ -132,7 +132,7 @@ The following config option are provided by the OpenHIM. All of these options ha
     "orchestrationTSBufferMillis": 100
   },
   "statsd": {
-    // Rather use a statsd server for reporting metrics
+    // Report application metrics and transaction metrics to a statsd server
```
Comment (hnnesv) [inline config/config.md]: aren't we getting rid of statsd though? or just api/statsd? I must admit to being a little confused how it's all fitting together! :) is [src/stats](https://github.com/jembi/openhim-core-js/blob/2cd15ba8d9e5e2147fc7427bb2d64f455d7cf004/src/stats.coffee) different to the metrics?

File: docs/dev-guide/api-ref.md [hunk 2]
```diff
@@ -629,90 +629,47 @@ The response will be an http status code of `201` if the channels were successfu
 
 ### Metrics resource
 
-#### Global Metrics
+This resource enables transaction metrics to be extracted from the OpenHIM in a flexible way. There are various forms of this endpoint depending on the format of the metrics that you want to get out. Metrics will only be returned for the channels that a the API user has access to.
```
Comment (hnnesv) [inline docs/dev-guide/api-ref.md]: typo: a the API user

File: config/config.md (hunk 1 above)
Comment (hnnesv) [inline config/config.md]: Aah ok think I answered my own question after reading through. So we're getting rid of statsd support on the metrics API, but keeping the existing reporting to statsd if someone wants to use it...

File: src/metrics.coffee [hunk 3]
```diff
@@ -1,356 +1,123 @@
...
+# @startDate {Date} a timestamp representing the end of the range to calculate
+#                   metrics to (required)
+# @transactionFilter {Object} a mongodb filter object to further restrict the
+#                             transactions collection (nullable)
+# @channelIDs {Array} an array of channel IDs as `ObjectID`s to filter by, if
+#                     not set all channels will be considered (nullable)
+# @timeSeries {String} one of 'minute', 'hour', 'day', 'week', 'month', 'year'.
+#                      If set the metrics will be grouped into a periods of the
+#                      stated duration, otherwise, metrics for the entire period
+#                      will be returned (nullabe)
```
Comment (hnnesv) [inline src/metrics.coffee]: typo: nullable

**Comment by hnnesv** [comment]
Looks good to me! Just some typos but good to merge after 👍 

I've been playing around with the API as well locally and it worked well!

File: config/config.md (hunk 1 above)
Comment (rcrichton) [inline config/config.md]: Yeah, exactly. I figured why not. We use statsd for other things already (application metrics), may as well not delete this. The metrics calculations were the headache anyways. What do you think? Agree?

**Comment by rcrichton** [comment]
@hnnesv awesome. Did you note the time taken to do the metrics calculations? For me I queried the last two years of transactions and it was still snappy. I think mongo seems to do a fairly good job.

**Comment by rcrichton** [comment]
@hnnesv I've fixed up the typos

File: config/config.md (hunk 1 above)
Comment (hnnesv) [inline config/config.md]: Yeah definitely, sounds good

**Comment by hnnesv** [comment]
Awesome! :) No I haven't tested with that much data, it was just a 100 or so over a couple days. Yeah I think aggregation should work well for us.

**Comment by hnnesv** [comment]
LGTM 👍

**Pull request merged**